File: Python/238_Product_of_Array_Except_Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        a = []
        b = [0]*n
        a.append(1)
        a.append(nums[0])
        b[n-1] = 1
        b[n-2] = nums[n-1]
        for i in range(2, n):
            a.append(nums[i-1]*a[i-1])
        for j in range(n-3, -1, -1):
            b[j] = b[j+1] * nums[j+1]
        for i in range(n):
            nums[i] = a[i] * b[i]
        return nums

File: C++/238_Product_of_Array_Except_Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> a(nums.size());
        vector<int> b(nums.size());
        vector<int> c(nums.size());
        a[0] = 1;
        for(int i = 1; i < nums.size(); i++){
            a[i] = a[i-1] * nums[i-1];
        }
        b[nums.size()-1] = 1;
        for(int i = nums.size() - 2; i >= 0; i--){
            b[i] = b[i+1] * nums[i+1];
        }
        for(int i = 0; i < nums.size(); i++){
            c[i] = a[i] * b[i];
        }
        return c;
    }
};